rclcpp/include/rclcpp/utilities.hpp

```diff
@@ -282,6 +313,113 @@ sub_will_underflow(const T x, const T y)
   return (y > 0) && (x < (std::numeric_limits<T>::min() + y));
 }
 
+/// Wrapper which will check for over and underflow of subtractions
+/**
+ * Function will check, if
+ * x - y
+ * is permissible. If not, it will throw an according exception
+ *
+ * \tparam T must be arithmetic
+ * \param x lhs of the subtraction
+ * \param y rhs of the subtraction
+ * \throws std::underflow_error if subtraction underflows
+ * \throws std::overflow_error if subtraction overflows
+ */
+template<typename T>
+void
+check_sub(const T x, const T y)
+{
+  static_assert(std::is_arithmetic<T>::value, "must be arithmetic");
+
+  if (sub_will_overflow(x, y)) {
+    const auto error_message = std::string("subtraction leads to overflow of ") +
+      std::string(typeid(x).name());
+    throw std::overflow_error(error_message);
+  }
+  if (sub_will_underflow(x, y)) {
+    const auto error_message = std::string("subtraction leads to underflow of ") +
+      std::string(typeid(x).name());
+    throw std::underflow_error(error_message);
+  }
+}
+
+/// Checks if a multiplication will lead to an overflow
+/**
+ * Function checks if
+ * res = x * y
+ * will lead to an overflow res, where res has the data type of x
+ *
+ * \tparam X data type of the left hand side
+ * \tparam Y data type of the right hand side
+ * \param x value of the left hand side
+ * \param y value of the right hand side
+ * \return true, if multiplication overflows
+ */
+template<typename X, typename Y>
```

Comment:
shouldn't this function also check for arithmetic types. Also, I am not sure if having two template types can potentially lead to UB when doing `X / Y`.

Reply:
The only ub I'm aware of during the division would be division by zero. This should'nt happend since we are checking beforehand if y is non-zero. What do you think?

Reply:
Ok, it was buggy as hell. Now it should be better